var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -76,6 +76,46 @@ class Trilinos(CMakePackage):
     variant('xsdkflags',    default=False,
             description='Compile using the default xSDK configuration')
 
+    # CUDA options (compare to kokkos/package.py)
+    variant('cuda', default=False, description="enable Cuda backend")
+    # kokkos has conflicts with cuda for kokkos@2.5:2.7, restrict this variant
+    # to @12.14.1 or later.
+    conflicts('+cuda', when='@:12.14.0')
+    conflicts('+cuda', when='~kokkos')
+    variant('kokkos_enable_cuda_uvm', default=False,
+            description="set force_uvm Kokkos CUDA option")
+    variant('kokkos_enable_rdc', default=False,
+            description="set rdc Kokkos CUDA option")
+    variant('kokkos_enable_cuda_lambda', default=False,
+            description="set enable_lambda Kokkos CUDA option")
+
+    # CUDA: Set the Host and GPU architecture variants
+    gpu_values = ('Kepler30', 'Kepler32', 'Kepler35', 'Kepler37',
+                  'Maxwell50', 'Maxwell52', 'Maxwell53',
+                  'Pascal60', 'Pascal61',
+                  'Volta70', 'Volta72')
+    host_values = ('AMDAVX', 'ARMv80', 'ARMv81', 'ARMv8-ThunderX',
```

Comment:
The user may run into problems if they're using an older compiler.  Not much we can do about that, but we should consider printing a warning message, or putting a comment in the code.